### PR TITLE
Feature/unity 1160 optional subsystem

### PR DESCRIPTION
**Pull request by BryceNewell-UL**
## Summary

Adding a settings file and using it to enable and disable the use for the LeapC subsystem. Settings file can be found in the base of the "Tracking" plugin.

## Contributor Tasks

_These tasks are for the merge request creator to tick off when creating a merge request._

- [ ] Pair review with a member of the QA team.
- [x] Add any release testing considerations to the MR for the next release.
- [ ] Check any relevant CHANGELOG files have been updated.
- [x] Ensure documentation requirements are met e.g., public API is commented.
- [x] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.
- [x] Add any relevant labels such as `breaking` to this MR.
- [ ] If this MR closes a Jira issue, make sure the fix version on the JIRA issue is set to the correct one.

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

[Use emojis in review threads to communicate intent and help contributors.](https://github.com/ultraleap/UnityPlugin/blob/develop/CONTRIBUTING.md#review-threads)

- [x] Code reviewed.
- [x] Non-code assets e.g. Unity assets/scenes reviewed.
- [x] Documentation has been reviewed. Includes checking documentation requirements are met and not missing e.g., public API is commented.
- [ ] Checked and agree with release testing considerations added to MR for the next release.

## Closes JIRA Issue

_If this MR closes any JIRA issues list them below in the form `Closes PROJECT-#`_

### CODE REVIEW

File: Packages/Tracking/Core/Runtime/Scripts/UltraleapSettings.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+        private void LeapSubSystemSection()
+        {
+            EditorGUILayout.Space(10);
+            EditorGUILayout.BeginVertical();
+            EditorGUILayout.LabelField("Leap XRHands Subsystem", EditorStyles.boldLabel);
+            
+            var readOnlyText = new GUIStyle(EditorStyles.textField);
+            readOnlyText.wordWrap = true;
+            EditorGUILayout.HelpBox("The Leap subsystem should be used when not using OpenXR for hand tracking inputs." +
```
Comment (MattGrayUL) [inline Packages/Tracking/Core/Runtime/Scripts/UltraleapSettings.cs:31]: Can this read:

> The Leap XRHands Subsystem passes hand input directly from the Leap Service to XRHands. If you are using OpenXR for hand input, do not enable this option. Instead, use the Hand Tracking Subsystem option in the OpenXR XR Plug-in Management settings.
> This option can not be toggled at runtime.

please?

File: Packages/Tracking/Core/Runtime/Scripts/UltraleapSettings.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            bool editorBool = settingsTarget.LeapSubsystemEnabled;
+            EditorGUILayout.BeginHorizontal();
+            settingsTarget.LeapSubsystemEnabled = EditorGUILayout.ToggleLeft("Enable the leap subsystem ", editorBool);
+            EditorGUILayout.EndHorizontal();
+            EditorGUILayout.EndVertical();
+        }
+    }
+
+
+    //[CreateAssetMenu(menuName = "Ultraleap/Settings/PluginSettings")]  /// comment out so the user cannot create these but may be needed for development
```
Comment (MattGrayUL) [inline Packages/Tracking/Core/Runtime/Scripts/UltraleapSettings.cs:44]: no longer needed?

File: Packages/Tracking/Core/Runtime/Scripts/XRHandsSubsystem/SubsystemStarter.cs [hunk 3]
```diff
@@ -11,17 +11,27 @@ public class SubsystemStarter
     {
         static XRHandSubsystem m_Subsystem = null;
         static XRHandProviderUtility.SubsystemUpdater updater = null;
+        static GameObject leapProviderGO = null;
+
+        static UltraleapSettings ultraleapSettings;
```
Comment (MattGrayUL) [inline Packages/Tracking/Core/Runtime/Scripts/XRHandsSubsystem/SubsystemStarter.cs:16]: This is only used in RunAfterSceneLoad so does not need to be a static here?

**Review by MattGrayUL — COMMENTED** [review]
This works great with my initial test in editor, need to test in builds etc still.

My only other initial thought is that we should have a quick menu option to select the Ultraleap Settings asset - in the bar at the top, the same as we have the "Ultraleap/Show Readme For Scene" option

File: Packages/Tracking/Core/Runtime/Scripts/UltraleapSettings.cs (hunk 2 above)
Comment (BryceNewell-UL) [reply Packages/Tracking/Core/Runtime/Scripts/UltraleapSettings.cs:44]: So this is the only way to create a new SO. Not needed for the user but might be useful if we ever need to delete the scriptable object during development this is the only way to get it back. 

Tempted to leave it in commented out for this reason

Comment (MattGrayUL) [reply Packages/Tracking/Core/Runtime/Scripts/UltraleapSettings.cs:44]: I think if we were to do that, we should just have an option to reset and some cached defaults - ideally users shouldn't be creating/deleting this asset

**Review by MattGrayUL — COMMENTED** [review]
The editor content will need to be wrapped in #if UNITY_EDITOR including the "using UnityEditor" or this content will not build

**Pull request merged**